exonum/src/api/public/system.rs

```diff
@@ -104,7 +125,14 @@ impl Api for SystemApi {
             }
         };
 
+        let _self = self.clone();
```

Comment:
Maybe `self_` instead of `_self`? Underscore prefix suppress "unused variable" warning.

Reply:
I've just replicated the surrounding code: https://github.com/exonum/exonum/blob/4f0f133f0/exonum/src/api/public/system.rs#L81

Reply:
I don't like the "surrounding code" either. :smile:

Reply:
Yep, but should I really do this in the current PR?

```text
$ git grep _self
exonum/src/api/private/system.rs:        let _self = self.clone();
exonum/src/api/private/system.rs:                    _self.peer_add(ip_str)?;
exonum/src/api/private/system.rs:                    _self.ok_response(&::serde_json::to_value("Ok").unwrap())
exonum/src/api/private/system.rs:        let _self = self.clone();
exonum/src/api/private/system.rs:            let info = _self.get_peers_info();
exonum/src/api/private/system.rs:            _self.ok_response(&::serde_json::to_value(info).unwrap())
exonum/src/api/private/system.rs:        let _self = self.clone();
exonum/src/api/private/system.rs:            let info = _self.get_network_info();
exonum/src/api/private/system.rs:            _self.ok_response(&::serde_json::to_value(info).unwrap())
exonum/src/api/public/blockhain_explorer.rs:        let _self = self.clone();
exonum/src/api/public/blockhain_explorer.rs:            let info = _self.get_blocks(count, latest, skip_empty_blocks)?;
exonum/src/api/public/blockhain_explorer.rs:            _self.ok_response(&::serde_json::to_value(info).unwrap())
exonum/src/api/public/blockhain_explorer.rs:        let _self = self.clone();
exonum/src/api/public/blockhain_explorer.rs:                    let info = _self.get_block(Height(height))?;
exonum/src/api/public/blockhain_explorer.rs:                    _self.ok_response(&::serde_json::to_value(info).unwrap())
exonum/src/api/public/system.rs:        let _self = self.clone();
exonum/src/api/public/system.rs:            let info = _self.get_mempool_info();
exonum/src/api/public/system.rs:            _self.ok_response(&::serde_json::to_value(info).unwrap())
exonum/src/api/public/system.rs:        let _self = self.clone();
exonum/src/api/public/system.rs:                    let info = _self.get_transaction(hash_str)?;
exonum/src/api/public/system.rs:                    result(&_self, &::serde_json::to_value(info).unwrap())
exonum/src/api/public/system.rs:        let _self = self.clone();
exonum/src/api/public/system.rs:            let info = _self.get_healthcheck_info();
exonum/src/api/public/system.rs:            _self.ok_response(&::serde_json::to_value(info).unwrap())
exonum/src/node/consensus.rs:            let hash = self.state.add_self_propose(propose);
exonum/src/node/state.rs:    pub fn add_self_propose(&mut self, msg: Propose) -> Hash {
```

Reply:
> but should I really do this in the current PR?

I don't care in which PR it would be done, if it would be done at all.
And I propose at least not to introduce one more `_self` in your changes.